datajunction-server/datajunction_server/models/node.py

```diff
@@ -739,20 +740,34 @@ def check_metric(self):
         tree = parse(self.query)
         if len(tree.select.projection) != 1:
             raise DJInvalidInputException(
-                "Metric queries can only have a single "
+                http_status_code=HTTPStatus.BAD_REQUEST,
+                message="Metric queries can only have a single "
                 f"expression, found {len(tree.select.projection)}",
             )
         projection_0 = tree.select.projection[0]
 
+        # must have an aggregation
         if (
             not hasattr(projection_0, "is_aggregation")
             or not projection_0.is_aggregation()  # type: ignore
         ):
             raise DJInvalidInputException(
-                f"Metric {self.name} has an invalid query, "
+                http_status_code=HTTPStatus.BAD_REQUEST,
+                message=f"Metric {self.name} has an invalid query, "
                 "should have a single aggregation",
             )
 
+        # must query from a single table
+        if (
+            len(tree.select.from_.relations) != 1
+            or tree.select.from_.relations[0].extensions
+        ):
+            raise DJInvalidInputException(
+                http_status_code=HTTPStatus.BAD_REQUEST,
+                message=f"Metric {self.name} selects from more than one table, "
+                "should only select from a single table",
```

Comment:
Should this message use the `table` terminology or would `node` be clearer? (Not sure as I'm not even sure users like `node`, but it fits well with what we call them)

Reply:
Agree with @shangyian . I think there is a chance we can win this "node vs table" so I would push on that concept where it fits well. The other DJ dag terms we can still keep under covers, like "bound dimensions" ;)

Reply:
Good catch, we should consistently use node so I'll update this.